cmd/acr/purge.go

```diff
@@ -264,10 +279,22 @@ func getMatchingRepos(ctx context.Context, repoNames []string, repoRegex string)
 
 // getRepositoryAndTagRegex splits the strings that are in the form <repository>:<regex filter>
 func getRepositoryAndTagRegex(filter string) (string, string, error) {
-	repoAndRegex := strings.Split(filter, ":")
+	// This only selects colons that are not apart of a non-capture group
+	// Note: regexp2 doesn't have .Split support yet, so we just replace the colon with another delimitter \r\n
+	// We choose \r\n since it is an escape sequence that cannot be a part of repo name or a tag
+	noncaptureGroupSupport := regexp2.MustCompile("(?<!\\(\\?):", defaultRegexpOptions)
```

Comment:
nit: It will be more readable with raw strings. 
```suggestion
	noncaptureGroupSupport := regexp2.MustCompile(`(?<!\(\?):`, defaultRegexpOptions)
```

Reply:
fixed